compiler/rustc_lexer/src/lib.rs

```diff
@@ -415,8 +415,16 @@ impl Cursor<'_> {
             }
 
             // Guarded string literal prefix: `#"` or `##`
-            '#' if matches!(self.first(), '"' | '#') => {
+            '#' if matches!(
+                (self.first(), self.second()),
+                //  #"         ##"          ###
+                ('"', _) | ('#', '"') | ('#', '#')
```

Comment:
*Miiiildly* concerned about the implicit interaction between parser and lexer with these now, given the new reserved syntax, but that goes beyond *this* PR and is more about the feature as a whole, and I'm not going to block that.